test/system/045-start.bats

```diff
@@ -99,4 +99,45 @@ load helpers
     run_podman rm -t 0 -f $ctrID $cname
 }
 
+@test "podman start again with lower ulimit -u" {
+    skip_if_not_rootless "tests ulimit -u changes in the rootless scenario"
+    skip_if_remote "test relies on control of ulimit -u (not possible for remote)"
+    # get current ulimit -u
+    nrpoc_limit=$(ulimit -Hu)
+
+    # create container
+    run_podman create $IMAGE echo "hello"
+    ctrID="$output"
+
+    # inspect
+    run_podman inspect $ctrID
+    assert "$output" =~ '"Ulimits": \[\]' "Ulimits has to be empty after create"
+
+    # start containers for the first time
+    run_podman start $ctrID
+    is "$output" "$ctrID"
+
+    # inspect
+    run_podman inspect $ctrID
+    assert "$output" =~ ${nrpoc_limit} "Soft limit has to be set to ulimit -u"
```

Comment:
> Also added a test under test/system that starts a container, lowers the ulimit and starts the container a second time. Couldn't get the assert statement properly escaped. It's currently only matching against the number of the ulimit. I would prefer to match against "Soft": $nproc_limit. How do I escape this properly? Could also use jq alternatively.

How about: 
```
podman inspect --format '{{range .HostConfig.Ulimits}}{{if eq .Name "RLIMIT_NPROC" }}{{.Soft}}:{{.Hard}}{{end}}{{end}}'
```
Then just use `assert "$output" == "${nrpoc_limit}:${nrpoc_limit}"`

Reply:
Nice, didn't think of `--format`, works!